src/common/types.ts

```diff
@@ -37,13 +37,12 @@ export interface Column<TRow> {
   filterRenderer?: React.ComponentType<FilterRendererProps<TRow, any>>;
 }
 
-export type CalculatedColumn<TRow> =
-  Column<TRow> & {
-    idx: number;
-    width: number;
-    left: number;
-    formatter: React.ComponentType<FormatterProps<TRow>>;
-  };
+export interface CalculatedColumn<TRow> extends Column<TRow> {
```

Comment:
Do we need to omit `Column<Row>.formatter`?

Reply:
Doesn't look like it, it's overriding both width and formatter accordingly.